packages/backend-common/src/tokens/types.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface TokenManager {
+  getServerToken: () => Promise<{ token: string }>;
```

Comment:
Bit torn on the names here, given that we're calling it a `TokenManager` I think `getToken` and `validateToken` might be the way to go. It does allow us to create a couple of layers of these in the future that collaborate, something like this:

```ts
const tokenManager = UnifiedTokenManager.create({
  userTokenManager: UserTokenManager.fromConfig(config),
  serverTokenManager: ServerTokenManager.fromConfig(config),
})
```

Or whatever flavor of that we might imagine :grin:

Thoughts? 😅

Reply:
Yep, makes sense! Done in 4b6de2f872.